### ISSUE TITLE
Only support  color_type = 2

### ISSUE THREAD

**Issue by esdrasedu** [SenecaSystems/gutenex#3]
`Gutenex.PDF.Builders.ImageBuilder.add_image_extra_object` only support color_type: 2, my png is 6

**Comment by tyre** [comment]
yes! I'm almost done with this one. I've been rewriting SenecaSystems/Imagineer to handle every color type.

**Comment by esdrasedu** [comment]
is possible convert png color_type 6 to color_type 2 using gimp?

**Comment by tyre** [comment]
You may be able to do it with imagemagick. The color type indicates which palettes to use. Color type 2 is for RGB. 6 means RGB plus an alpha value. So you could convert them, but you would lose the alpha (if this matters to you.)

If it _does_ matter, I need to add support for all image types. That is something I should do anyway and will work on in parallel with Imagineer

**Comment by tyre** [comment]
Okay, I've just merged in changes to [SenecaSystems/Imagineer](https://github.com/SenecaSystems/imagineer) to handle reading and writing for all color types.

Next up is supporting all of them in gutenex which should, in theory, not be impossible
